server/src/instant/auth/oauth.clj

```diff
@@ -131,19 +131,24 @@
 
 (defn fetch-discovery [endpoint]
   (let [resp (clj-http/get endpoint {:throw-exceptions false
-                                     :as :json})]
+                                     :as :json
+                                     ;; for https://account.apple.com/.well-known/openid-configuration
+                                     :headers {"User-Agent" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.6 Safari/605.1.15"}})]
     (if (clj-http/success? resp)
       {:date (Instant/now)
        :data (:body resp)}
-      (ex/throw-oauth-err! "Unable to fetch OAuth configuration."))))
+      (ex/throw+ {::ex/type    ::ex/oauth-error
+                  ::ex/message "Unable to fetch OAuth configuration"
+                  :status      (:status resp)
+                  :body        (:body resp)}))))
```

Comment:
I'm a little antsy about putting `body` in an instant exception on the off chance we don't sanitize properly and expose the body to the user. Maybe before the `throw-oauth-err!`, we could add an exception trace?

```clj
(tracer/record-exception-span! (ex-info "Error fetching discovery"
                                        {:status (:status resp)
                                         :body (:body resp)
                                         :endpoint endpoint}
                                        {:name "oauth/fetch-discovery-error"})
```

Reply:
Done